src/nodes/NodeManager.ts

```diff
@@ -474,45 +536,48 @@ class NodeManager {
           nodeId,
           nodeAddress,
           timeout,
+          options,
         );
       } else {
         this.logger.debug(
           `Bucket was full and blocking was false, adding ${nodesUtils.encodeNodeId(
             nodeId,
           )} to queue`,
         );
-        // Re-attempt this later asynchronously by adding the the queue
-        this.queue.push(() =>
-          this.setNode(nodeId, nodeAddress, true, false, timeout),
+        // Re-attempt this later asynchronously by adding to the scheduler
+        await this.taskManager.scheduleTask(
+          {
+            handlerId: this.setNodeHandlerId,
+            parameters: [nodesUtils.toString(), nodeAddress, timeout],
+            path: ['setNode'],
+            lazy: true,
+          },
+          tran,
         );
       }
     }
   }
 
+  // FIXME: make cancellable
   private async garbageCollectOldNode(
     bucketIndex: number,
     nodeId: NodeId,
     nodeAddress: NodeAddress,
     timeout?: number,
+    options: { signal?: AbortSignal } = {},
   ) {
```

Comment:
The garbage collection mechanism is changing to become like a while loop that goes over all the existing and pending nodes. We will need a `Semaphore` exposed in the `async-locks` library so the while loop can iterate over the combined bucket upon acquiring a lock. The semaphore limit is set to the k parallel constant.

This should allow us to collapse the existing and pending nodes into a single bucket with the remaining nodes.

Reply:
The maximum number of pings that could occur is the 20 bucket limit, but the minimum is however many nodes there are in the pending bucket partition multiplied by the k parallel constant.

Reply:
Actually max is 20, min will be 3. The true number of pings will be somewhere in between.

Reply:
Garbage collection has been updated. As a result, `setNode` is no longer has optional blocking or abortion with signal.